default/python/AI/ColonisationAI.py

```diff
@@ -1280,6 +1281,15 @@ def send_colony_ships(colony_fleet_ids, evaluated_planets, mission_type):
                 universe.getSystem(sys_id), aistate.systemStatus[sys_id]['monsterThreat']))
             already_targeted.append(planet_id)
             continue
+
+        # do not try to colonize in unexplored systems - there often are guards!
+        # Note that we can have vision of a planet in an unexplored system due to e.g. some species effects
+        has_explored_system = get_partial_visibility_turn(sys_id) > 0
+        if not has_explored_system:
```

Comment:
I am not sure who has, maybe we can rename it to the system_is_explored?

```suggestion
        system_is_explored = get_partial_visibility_turn(sys_id) > 0
        if not system_is_explored :
```

As an alternative solution, we could put the code, TODO, and comment to the separated function:

```python
if system_may_be_guarded():
```